starter-code/routes/logIn.js

```diff
@@ -0,0 +1,45 @@
+const express = require("express");
+const router = express.Router();
+
+const User = require("../user/user.js");
+
+const bcrypt         = require("bcrypt");
+const bcryptSalt     = 10;
+
+
+//GET logIn Page
+router.get("/logIn", function(req, res, next){
+  res.render("logIn");
+});
+
+router.post("/logIn", (req, res, next) => {
+  let username = req.body.username;
+  let password = req.body.password;
+  let salt     = bcrypt.genSaltSync(bcryptSalt);
+  let hashPass = bcrypt.hashSync(password, salt);
+
+  if (username === "" || password === "") {
+    console.log("please provide a username and password");
+  } else {
+    User.findOne({username}, "username password", (error, user) => {
+      if (error) {
+        throw error;
+      } else if(user.username === username && user.password === hashPass) {
+        res.render("welcome");
```

Comment:
so, we have a login at this point

you need to store the retrieved user in the `req.session` object

```
req.session.currentUser = user;
```

so that in other routes we can check if user is logged in or not

note: depends on having setup correctly the session middleware in `app.js`